### PR TITLE
Add calculations of precision and recall on a particular label

### PR DESCRIPTION
**Pull request by gabechu**
### Context
In PII recognition, we have two approaches to score the system. If we do NOT care about the entity type, we will run a binary classification to evaluate the performance of boundary detection on all entities. If we DO care about the entity type, we will run a multi-class classification to evaluate the performance of boundary detection on different entity types.

### Description
Created two functions: one for recall calculation and the other for precision calculation. The functions work for both binary and multi-class labels and support integer and string labels or mix of both as inputs.

#### Checklist
- [x] Added **tests** for changed code.

### CODE REVIEW

File: pii_recognition/evaluation/metrics_test.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import numpy as np
-from .metrics import compute_f_beta
+from numpy.testing import assert_almost_equal
+
+from .metrics import compute_f_beta, compute_label_precision, compute_label_recall
 
 
 def test_compute_f_beta():
+    # TODO: break into separate test functions
```
Comment (gabechu) [inline pii_recognition/evaluation/metrics_test.py:8]: Will resolve in a separate PR.

Comment (chausler) [inline pii_recognition/evaluation/metrics_test.py:8]: why not use `sklearn.metrics.fbeta_score`?

**Review by chausler — APPROVED** [review]

File: pii_recognition/evaluation/metrics_test.py (hunk 1 above)
Comment (gabechu) [reply pii_recognition/evaluation/metrics_test.py:8]: Make sense. Will put another PR switching to sklearn function. Back then I was thinking of reducing the dependencies when implementing the evaluation make it not relying on many libraries, but since we are going to use sklearn anyway, it won't make any difference.

Comment (gabechu) [reply pii_recognition/evaluation/metrics_test.py:8]: Had another look, might not be able to switch to fbeta_score. First, in previous research where it gets applied we do not have `y_true` and `y_pred` arrays instead we have `Counter` for calculation. Second, there are customised conditions, depending on the values of precision and recall strategies vary.
